Improving nav bar

DoctorDib opened the pull request:
Changes:
 - Added sticky nav bar feature for easier access on all pages.

DoctorDib committed:
Adding a sticky header

Changes:
 - Added a sticky header to make the nav bar / header easier to access

DoctorDib committed:
Fixing lint

Changes:
 - Removed all new lint errors

BenRGH commented on client/src/Components/Header/index.tsx:
```diff
@@ -22,7 +22,14 @@ const LogoButton = withStyles({
+    const handleScroll = () => setScrolled(window.scrollY > 100);
+
+    useEffect(() => {
+        window.addEventListener('scroll', handleScroll);
```
You need to add a return statement that removes the listener or we get memory leaks

BenRGH replied:
Also add an empty array as the second arg for the useEffect or it'll rerun every render

DoctorDib replied:
I think your first comment is related to the second. I did forget to add the empty array which I have just changed

BenRGH replied:
Nah you still need the return https://stackoverflow.com/a/55566585/9982232

DoctorDib replied:
https://codedaily.io/tutorials/72/Creating-a-Reusable-Window-Event-Listener-Hook-with-useEffect-and-useCallback

DoctorDib committed:
Removing re-render issue

DoctorDib committed:
Adding return

Changes:
 - Adding return to useHooks - https://codedaily.io/tutorials/72/Creating-a-Reusable-Window-Event-Listener-Hook-with-useEffect-and-useCallback

DoctorDib committed:
Whoopsies
